plugins/net_plugin/net_plugin.cpp

```diff
@@ -2519,6 +2560,58 @@ namespace eosio {
       } );
    }
 
+   void dispatch_manager::bcast_hs_proposal_msg(const hs_proposal_message_ptr& msg) {
+      if( my_impl->sync_master->syncing_from_peer() ) return;
+      hs_proposal_message& msg_val = *(msg.get());
+      my_impl->connections.for_each_block_connection( [&msg_val]( auto& cp ) {
+         if( !cp->current() ) return true;
+         cp->strand.post( [cp, msg_val]() {
+            if (cp->protocol_version >= proto_instant_finality)
+               cp->enqueue( msg_val );
+         });
+         return true;
+      } );
+   }
+
+   void dispatch_manager::bcast_hs_vote_msg(const hs_vote_message_ptr& msg) {
+      if( my_impl->sync_master->syncing_from_peer() ) return;
```

Comment:
We will likely need a new implementation for `syncing_from_peer`. Since `lib` will be so close to `head` the `net_plugin` will need to be protected from dropping into lib catchup mode when it has simply not moved its lib yet. This might be as simple as a hard-coded delta from peer head before assuming it is syncing (or should sync).